lib/Wikia/src/CircuitBreaker/LocalCircuitBreaker.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+
+namespace Wikia\CircuitBreaker;
+
+
+use Ackintosh\Ganesha;
+use Ackintosh\Ganesha\Builder;
+use Wikia\CircuitBreaker\Storage\APCUAdapter;
+use Wikia\Logger\WikiaLogger;
+
+class LocalCircuitBreaker implements CircuitBreaker {
+
+	/** @var Ganesha */
+	private $ganesha;
+
+	public function __construct() {
+		$this->ganesha = Builder::build( [
+			'failureRateThreshold' => 50,
+			'adapter' => new APCUAdapter(),
+			'intervalToHalfOpen' => 5,
+			'minimumRequests' => 4,
+			'timeWindow' => 30,
+		] );
+
+		$this->ganesha->subscribe( function ( $event, $service, $message ) {
+			switch ( $event ) {
+				case Ganesha::EVENT_TRIPPED:
+					WikiaLogger::instance()
+						->error( "Circuit open! It seems that a failure has occurred in {$service}. {$message}." );
```

Comment:
It is more usable to log constant message with variable context:
```suggestion
						->error( "Circuit open! It seems that a failure has occurred in", ["service" => $service, "error"-> $message,] );
```

Reply:
The same applies to the following log statements.